packages/material-renderers/test/renderers/MaterialArrayLayout.test.tsx

```diff
@@ -629,4 +634,141 @@ describe('Material array layout', () => {
     const noDataLabel = wrapper.find('div>div>p').text();
     expect(noDataLabel.includes('Translated')).toBeTruthy();
   });
+
+  it('should have a tooltip for add button', () => {
+    wrapper = checkTooltip(
+      wrapper,
+      (wrapper) => wrapper.find(ArrayLayoutToolbar),
+      ArrayTranslationEnum.addTooltip,
+      {
+        id: 'tooltip-add',
+      }
+    );
+  });
+  it('should have a translatable tooltip for add button', () => {
+    wrapper = checkTooltipTranslation(
+      wrapper,
+      (wrapper) => wrapper.find(ArrayLayoutToolbar),
+      {
+        id: 'tooltip-add',
+      }
+    );
+  });
+
+  it('should have a tooltip for delete button', () => {
+    wrapper = checkTooltip(
+      wrapper,
+      (wrapper) => wrapper.find('Memo(ExpandPanelRendererComponent)').at(0),
+      ArrayTranslationEnum.removeTooltip,
+      {
+        id: 'tooltip-remove',
+      }
+    );
+  });
+  it('should have a translatable tooltip for delete button', () => {
+    wrapper = checkTooltipTranslation(
+      wrapper,
+      (wrapper) => wrapper.find('Memo(ExpandPanelRendererComponent)').at(0),
+      {
+        id: 'tooltip-remove',
+      }
+    );
+  });
+
+  it('should have a tooltip for up button', () => {
+    wrapper = checkTooltip(
+      wrapper,
+      (wrapper) => wrapper.find('Memo(ExpandPanelRendererComponent)').at(0),
+      ArrayTranslationEnum.up,
+      {
+        id: 'tooltip-up',
+      }
+    );
+  });
+  it('should have a translatable tooltip for up button', () => {
+    wrapper = checkTooltipTranslation(
+      wrapper,
+      (wrapper) => wrapper.find('Memo(ExpandPanelRendererComponent)').at(0),
+      {
+        id: 'tooltip-up',
+      }
+    );
+  });
+
+  it('should have a tooltip for down button', () => {
+    wrapper = checkTooltip(
+      wrapper,
+      (wrapper) => wrapper.find('Memo(ExpandPanelRendererComponent)').at(0),
+      ArrayTranslationEnum.down,
+      {
+        id: 'tooltip-down',
+      }
+    );
+  });
+  it('should have a translatable tooltip for down button', () => {
+    wrapper = checkTooltipTranslation(
+      wrapper,
+      (wrapper) => wrapper.find('Memo(ExpandPanelRendererComponent)').at(0),
+      {
+        id: 'tooltip-down',
+      }
+    );
+  });
 });
+
+function checkTooltip(
+  wrapper: ReactWrapper<any, any>,
+  wrapperFindFunction: (
+    wrapper: ReactWrapper<any, any>
+  ) => ReactWrapper<any, any>,
+  tooltipEnum: ArrayTranslationEnum,
+  tooltipFilter: EnzymePropSelector | string
+) {
```

Comment:
Please type the return type of this function explicitly.

Reply:
Done